### PR TITLE
Register payformsg interfaces

### LINKED ISSUE #1: Replace dependency on official tendermint/cosmos libraries with our forks
## Feature description 
To ensure compatibility with our Celestia chain, we will need to depend on our forks of [tendermint](https://github.com/celestiaorg/celestia-core) and [cosmos-sdk](https://github.com/celestiaorg/cosmos-sdk), so that all transactions are handled correctly.

## Implementation proposal
Instead of making change on [juno](https://forbole/juno) and replace all imports with ours, and related work. For now, let's replace the dependencies directly in the current repository with ours. It's easier and more maintainable.

### PR DESCRIPTION
**Pull request by rach-id**
- Adds dependency to Celestia forks
- Registers `PayForMsg` interface to be decoded correctly

Closes: https://github.com/celestiaorg/bdjuno/issues/1, https://github.com/celestiaorg/bdjuno/issues/3 and https://github.com/celestiaorg/bdjuno/issues/4
Contains: https://github.com/celestiaorg/bdjuno/pull/2

### CODE REVIEW

File: types/config/encoding.go [hunk 1]
```diff
@@ -12,6 +13,7 @@ func MakeEncodingConfig(managers []module.BasicManager) func() params.EncodingCo
 		encodingConfig := params.MakeTestEncodingConfig()
 		std.RegisterLegacyAminoCodec(encodingConfig.Amino)
 		std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+		paymenttypes.RegisterInterfaces(encodingConfig.InterfaceRegistry)
```
Comment (evan-forbes) [inline types/config/encoding.go:16]: :+1:

File: go.mod [hunk 2]
```diff
@@ -24,4 +25,8 @@ replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alp
 
 replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
-replace github.com/tendermint/tendermint => github.com/forbole/tendermint v0.34.13-0.20210820072129-a2a4af55563d
+replace github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v0.34.14-celestia.0.20220119223021-e54097114e14
+
+replace github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v0.44.1-celestia
+
+replace github.com/forbole/juno/v2 v2.0.0-20211221122008-f95aacf17add => github.com/celestiaorg/juno/v2 v2.1.0-celestia
```
Comment (evan-forbes) [inline go.mod]: super minor optional nit: we might want to format these in a more normal way similar to https://github.com/celestiaorg/celestia-app/blob/2015a0cac19c841d5971c1155c53fa2b9f736d6b/go.mod#L125-L131

**Review by evan-forbes — APPROVED** [review]
nice work debugging! :+1: LGTM

File: go.mod (hunk 2 above)
Comment (rach-id) [reply go.mod]: done, thanks

**Pull request merged**